### PR TITLE
TASK: Node type manager minor refinement

### PR DESCRIPTION
**Pull request by mhsdesign**
- Declare @api in NodeTypeManager and adjust code style
- Remove dead code which was not reused
- Remove redundant string cast (->value) when calling the NodeTypeManager

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.Core/Classes/Feature/NodeAggregateCommandHandler.php [hunk 1]
```diff
@@ -181,72 +181,4 @@ public function withoutAncestorNodeTypeConstraintChecks(\Closure $callback): voi
 
         $this->ancestorNodeTypeConstraintChecksEnabled = $previousAncestorNodeTypeConstraintChecksEnabled;
     }
-
-    /**
-     * @todo perhaps reuse when ChangeNodeAggregateType is reimplemented
-     */
-    protected function checkConstraintsImposedByAncestors(
```
Comment (mhsdesign) [inline Neos.ContentRepository.Core/Classes/Feature/NodeAggregateCommandHandler.php:188]: surprise: it was not.

File: Neos.ContentRepository.Core/Classes/NodeType/NodeTypeManager.php [hunk 2]
```diff
@@ -24,24 +24,27 @@
 
 /**
  * Manager for node types
- * @api
+ * @api Note: The constructor is not part of the public API
  */
-class NodeTypeManager
+final class NodeTypeManager
```
Comment (mhsdesign) [inline Neos.ContentRepository.Core/Classes/NodeType/NodeTypeManager.php:29]: we dont mock this anymore and thus we dont need it to be non-final

File: Neos.ContentRepository.Core/Classes/Feature/Common/ConstraintChecks.php [hunk 3]
```diff
@@ -100,7 +100,7 @@ protected function requireDimensionSpacePointToExist(DimensionSpacePoint $dimens
     protected function requireNodeType(NodeTypeName $nodeTypeName): NodeType
     {
         try {
-            return $this->getNodeTypeManager()->getNodeType($nodeTypeName->value);
+            return $this->getNodeTypeManager()->getNodeType($nodeTypeName);
```
Comment (mhsdesign) [inline Neos.ContentRepository.Core/Classes/Feature/Common/ConstraintChecks.php:103]: the `->value` cast ist just unnecessary as we _do_ accept the dto direct and should use it for type safety

**Review by ahaeslich — APPROVED** [review]
by 👀

**Comment by kitsunet** [comment]
wow, wow, wow, single approval merge ;) Not nice. But post +1 from me

**Comment by mhsdesign** [comment]
yeah for minor 9.0 dev things i / we have been doing this ...

but you as rm have a say in this as well, id be open to discuss our strategies here, but in my opinion we should move fast at this stage with non critical things. Its just tooooo hard to get reviews. And while its easier to get reviews for little things like these, the reviews _should_ rather focus on the bigger things. Thats why i try to have as few mini noise prs open as possible and only open them if i stumble upon this thing multiple times in a row :D

But lets discuss this in a rm meeting?

**Pull request merged**